### PR TITLE
feat: Add advice over empty valued <option> in required <select>

### LINKED ISSUE #404: `<select required>` should start with an empty value `option`
Sources:
— https://twitter.com/WebPlatformNews/status/1346492730223058946
— https://html.spec.whatwg.org/multipage/form-elements.html#placeholder-label-option
— https://validator.w3.org/nu/?showsource=yes&doc=https%3A%2F%2Fjsbin.com%2Ftozopid%2Fquiet

First idea:

```css
select[required]:not([multiple]) option:first-child:not([value=""]):not([value=" "]) {
  …
}
```

### PR DESCRIPTION
**Pull request by Dionakra**
Hi, 

I have added a first version of the check in the project. As it needed several translations, I added the keyword for it in every `locales/*.json` file at least in plain English as, if I didn't, it wouldn't compile.

Should we give the English translations where we don't know the real translation or shall we use Deepl, Google Translate or some kind of tool alike?

I also added all the `css` and `sass` files to the PR as they are mandatory for translations.

Any change needed just tell me :smile: 

PS: I also added the Spanish language to the languages dropdown so, when the doc is updated and uploaded, we can use it through the bookmarklet.

---

Closes #404

### CODE REVIEW

File: locales/en.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
   "advice:table-presentation": "Layout table",
   "advice:no-href": "Link without target",
   "advice:no-https": "No HTTPS",
+  "advice:select-option-not-empty": "<select> should have empty [value] on first <option>",
```
Comment (ffoodd) [inline locales/en.json:29]: We need to mention that it only applies to `required` select:
```suggestion
  "advice:select-option-not-empty": "<select required> should start with <option value="">",
```

And I'd go for code-ish content, since the docs are here to explain messages.

File: locales/fr.json [hunk 2]
```diff
@@ -26,6 +26,7 @@
   "advice:table-presentation": "Tableau de mise en forme",
   "advice:no-href": "Lien sans cible",
   "advice:no-https": "Pas de HTTPS",
+  "advice:select-option-not-empty": "<select> should have empty [value] on first <option>",
```
Comment (ffoodd) [inline locales/fr.json:29]: Translating my suggested english message to french:
```suggestion
  "advice:select-option-not-empty": "Un <select required> devrait débuter par <option value="">",
```

File: sass/themes/_advice.scss [hunk 3]
```diff
@@ -1,6 +1,41 @@
 @charset 'UTF-8';
 
 
+/*doc
+---
+title: "Required <select> should have first <option> with empty [value]"
```
Comment (ffoodd) [inline sass/themes/_advice.scss:6]:
```suggestion
title: "<select required> should start with an empty <option>'s [value]"
```

File: sass/themes/_advice.scss [hunk 4]
```diff
@@ -1,6 +1,41 @@
...
+select[required]:not([multiple])[size="1"],
+select[required]:not([multiple]):not([size])
+```
+
+##Test
+```html_example
+<select name="slices" required>
+  <option value="1">Cheese</option>
+  <option value="2">Salami</option>
+</select><span>†</span>
```
Comment (ffoodd) [inline sass/themes/_advice.scss:30]: I like your example :laughing:

File: sass/themes/_advice.scss [hunk 5]
```diff
@@ -1,6 +1,41 @@
...
 
+/*doc
+---
+title: "Required <select> should have first <option> with empty [value]"
+name: empty-option-on-select
+category: Advices
+---
+
+##Description
+The first child `option` element of a `select` element with a `[required]` attribute, and without a `[multiple]` attribute, and without a `[size]` attribute whose value is greater than 1, must have either an empty value attribute, or must have no text content. Consider either adding a placeholder option label, or adding a size attribute with a value equal to the number of option elements.
```
Comment (ffoodd) [inline sass/themes/_advice.scss:12]: It's a bit too complex as a sentence IMHO.

```suggestion
A `<select required>` element, which isn't neither `[multiple]` nor `[size]` greater than `1`, should start with an empty `<option>`. That being said, you may use a placeholder content for this option but must ensure to use an empty `[value]` attribute; or set a `[size]` attributes to the `<select>`, which value should equal to the number of `<option>`s.
```

What's your opinion?

File: sass/themes/_advice.scss [hunk 6]
```diff
@@ -1,6 +1,41 @@
...
 
 
+/*doc
+---
+title: "Required <select> should have first <option> with empty [value]"
+name: empty-option-on-select
+category: Advices
+---
+
+##Description
```
Comment (ffoodd) [inline sass/themes/_advice.scss:11]: Please add space between `##` and titles.

**Review by ffoodd — CHANGES_REQUESTED** [review]

**Comment by ffoodd** [comment]
Thanks, this is awesome! I made a few comments, mostly wording :+1:

File: sass/themes/_advice.scss (hunk 5 above)
Comment (Dionakra) [reply sass/themes/_advice.scss:12]: I agree, I just copied and pasted from one of the links to keep the description as close to the _official_ source, but I agree that is a bit confusing, at least for non-native English speakers like me. I struggled a bit trying to understand it at first, but your description is much better :smile:

File: locales/en.json (hunk 1 above)
Comment (Dionakra) [reply locales/en.json]: Done, only change I did was to escape the double quotes.

File: locales/fr.json (hunk 2 above)
Comment (Dionakra) [reply locales/fr.json]: Done! Same as before, only change I did was to escape the double quotes.

File: sass/themes/_advice.scss [hunk 7]
```diff
@@ -1,6 +1,41 @@
...
 
+/*doc
+---
+title: "<select required> should start with an empty <option>'s [value]"
+name: empty-option-on-select
+category: Advices
+---
+
+## Description
+A `<select required>` element, which isn't neither `[multiple]` nor `[size]` greater than `1`, should start with an empty `<option>`. That being said, you may use a placeholder content for this option but must ensure to use an empty `[value]` attribute; or set a `[size]` attributes to the `<select>`, which value should equal to the number of `<option>`s.
```
Comment (ffoodd) [inline sass/themes/_advice.scss:12]:
```suggestion
A `<select required>` element, which isn't `[multiple]` and whose `[size]` isn't greater than `1`, should start with an empty `<option>`. That being said, you may use a placeholder content for this option but must ensure to use an empty `[value]` attribute; or set a `[size]` attribute to the `<select>`, which value should equal to the number of `<option>`s.
```

**Review by ffoodd — CHANGES_REQUESTED** [review]
My bad, made a few errors in my grammar!

**Review by ffoodd — APPROVED** [review]
